src/legacy/components/ExplorerLink/index.tsx

```diff
@@ -31,31 +34,36 @@ export type Props = PropsWithId | PropsWithoutId
  * @param props
  */
 export function ExplorerLink(props: Props) {
-  const { chainId: _chainId } = useWalletInfo()
-  const chainId = _chainId || props.defaultChain
+  const { chainId, account } = useWalletInfo()
 
-  if (!chainId) {
-    return null
-  }
+  if (!account) return null
+
+  const { isComposableOrder = false } = props
+  const url = getUrl(chainId, isComposableOrder, account, props)
+
+  if (!url) return null
 
-  const url = getUrl(chainId, props)
   const { className } = props
 
-  const linkContent = getContent(chainId, props)
+  const linkContent = getContent(chainId, isComposableOrder, props)
   return (
     <ExternalLink className={className} href={url}>
       {linkContent}
     </ExternalLink>
   )
 }
 
-function getUrl(chainId: SupportedChainId, props: Props) {
+function getUrl(chainId: SupportedChainId, isComposableOrder: boolean, account: string, props: Props) {
   const { type } = props
 
   if (type === 'cow-explorer-home') {
     return getExplorerBaseUrl(chainId)
   }
 
+  if (isComposableOrder) {
+    return getSafeWebUrl(chainId, account, props.id)
+  }
```

Comment:
Shouldn't this be part of the type instead of a new parameter?
Like, `safe-url` or something like that.
I don't think the Explorer url needs to know about a particularity of the order, just how the link should be built.
Also, we could show the Safe link for other stuff rather than just composable orders

Reply:
Great suggestion, thanks! Refactored